src/components/Application/Navigation.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import { Button, Segment, Sticky } from 'semantic-ui-react'
+import {
+  MethodRevalidate,
+  MethodToCallProps,
+  SectionAndPage,
+  SectionDetails,
+  SectionsStructureNEW,
+} from '../../utils/types'
+import strings from '../../utils/constants'
+import { useRouter } from '../../utils/hooks/useRouter'
+
+interface NavigationProps {
+  current: SectionAndPage
+  sections: SectionsStructureNEW
+  serialNumber: string
+  requestRevalidation: MethodRevalidate
+}
+
+const Navigation: React.FC<NavigationProps> = ({
+  current,
+  sections,
+  serialNumber,
+  requestRevalidation,
+}) => {
+  const { push } = useRouter()
+
+  const currentSectionDetails = sections[current.sectionCode].details
+
+  const nextSection = (): SectionDetails | null => {
+    const nextSections = Object.values(sections)
+      .filter(({ details: { index } }) => index > currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => aIndex - bIndex)
+    return nextSections.length > 0 ? nextSections[0].details : null
+  }
+
+  const previousSection = (): SectionDetails | null => {
+    const previousSections = Object.values(sections)
+      .filter(({ details: { index } }) => index < currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => bIndex - aIndex)
+    return previousSections.length > 0 ? previousSections[0].details : null
+  }
+
+  const isFirstPage = current.pageNumber - 1 === 0 && previousSection() == null
+  const isLastPage =
+    current.pageNumber + 1 > currentSectionDetails.totalPages && nextSection() == null
+
+  const getPreviousSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber > 1) return { sectionCode, pageNumber: pageNumber - 1 }
+    return {
+      sectionCode: (previousSection() as SectionDetails).code,
+      pageNumber: (previousSection() as SectionDetails).totalPages,
+    }
+  }
+
+  const getNextSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber < currentSectionDetails.totalPages)
+      return { sectionCode, pageNumber: pageNumber + 1 }
+    return {
+      sectionCode: (nextSection() as SectionDetails).code,
+      pageNumber: 1,
+    }
+  }
+
+  const sendToPage = (sectionPage: SectionAndPage) => {
+    const { sectionCode, pageNumber } = sectionPage
+    push(`/applicationNEW/${serialNumber}/${sectionCode}/Page${pageNumber}`)
+  }
+
+  const previousButtonHandler = (_: any) => {
```

Comment:
You don't need any parameter here, I think. Can just do:
`const previousButtonHandler = () => {`

Reply:
But then instead of `onClick={previousButtonHandler}` I would need to do `onClick={() => previousButtonHandler}`. Preference?

Reply:
You don't have to do that, it works just fine with no parameter and your original `onClick`.

Reply:
>  You only need to do the other type of `onClick` when you need to pass a parameter from the onClick

Speaking of which -- I think the nicest way would be to have a single handler function and you just send a parameter (the page offset -- see below comment) to tell it whether to go back or fowards.

So next button would have `onClick={() => navButtonHandler(1)}` and back button would have `onClick={() => navButtonHandler(-1)}`

Reply:
I think that's good for more generic buttons, but here the Previous is very simple compared with Next, and I choose to just do the special validation on the nextButtonHandler

Reply:
Sure, but you still don't need the `_:any` parameters in `const previousButtonHandler = () => {...`

Reply:
Ok. Great. I'll do this change now. : )